spec/models/service_provider_spec.rb

```diff
@@ -94,4 +94,40 @@
       end
     end
   end
+
+  describe '#live?' do
+    context 'when the service provider is not approved' do
+      it 'returns false' do
+        sp = create(:service_provider, issuer: 'foo')
+
+        expect(sp.live?).to be false
+      end
+    end
+
+    context 'when the service provider is approved but not active' do
+      it 'returns false' do
+        sp = ServiceProvider.from_issuer('http://localhost:3000')
+        sp.update(active: false)
+
+        expect(sp.live?).to be false
```

Comment:
I believe it's because we want to have the ability to approve SPs once we go live rather than automatically allow any SP that gets published on the dashboard. Currently, `live?` is not being used, but we want to be able to use it in the future. https://github.com/18F/identity-idp/blob/master/app/services/saml_request_validator.rb#L31